src/analytics.js

```diff
@@ -160,6 +166,54 @@ export class Analytics {
 			});
 	}
 
+	_attachExceptionTracker() {
+		if (!this._options.exceptionTracking.enabled) {
+			return;
+		}
+
+		let options = this._options;
+		let existingWindowErrorCallback = window.onerror;
+
+		window.onerror = function (errorMessage, url, lineNumber, columnNumber, errorObject) {
+			// Send error details to Google Analytics, if library has loaded.
+			if (typeof ga === 'function') {
+				let exceptionDescription;
+				if (errorObject != undefined && typeof errorObject.message != undefined) {
+					exceptionDescription = errorObject.message;
+				} else {
+					exceptionDescription = errorMessage;
+				}
+
+				exceptionDescription += " @ " + url;
+				// Include additional details if available.
+				if (lineNumber != undefined && columnNumber != undefined) {
+					exceptionDescription += ":" + lineNumber + ":" + columnNumber;
+				}
+
+				let exOptions = {
+					exDescription: exceptionDescription,
+					exFatal: false
```

Comment:
Maybe in a future enhancement the user can pass certain exception types in the options property that determine whether the error should be considered fatal.